app/src/main/java/org/openedx/app/MainFragment.kt

```diff
@@ -64,6 +67,13 @@ class MainFragment : Fragment(R.layout.fragment_main) {
         viewModel.isBottomBarEnabled.observe(viewLifecycleOwner) { isBottomBarEnabled ->
             enableBottomBar(isBottomBarEnabled)
         }
+
+        arguments?.let {
+            it.getString(ARG_COURSE_ID, null)?.apply {
+                router.navigateToCourseDetail(parentFragmentManager, this)
+            }
+            it.putString(ARG_COURSE_ID, null)
+        }
```

Comment:
We can change `arguments?.let {}` to `requireArguments` if we always create MainFragment via newInstance()